resource/gomega.go

```diff
@@ -95,6 +106,29 @@ func matcherToGomegaMatcher(matcher interface{}) (types.GomegaMatcher, error) {
 	}
 }
 
+func mapToGomega(value interface{}) (subMatchers []types.GomegaMatcher, err error) {
+	valueI, ok := value.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Matcher expected map, got: %t", value)
+	}
+
+	var key, val types.GomegaMatcher
+	for k, v := range valueI {
+		key, err = matcherToGomegaMatcher(k)
```

Comment:
What does `matcherToGomegaMatcher()` on the key provide?

Can you give an example of this usage?